src/wallet/wallet.cpp

```diff
@@ -169,6 +169,13 @@ std::shared_ptr<CWallet> LoadWallet(interfaces::Chain& chain, const std::string&
     return LoadWallet(chain, WalletLocation(name), error, warnings);
 }
 
+bool CWallet::AbortRescan()
```

Comment:
Looks like you could have a race in this method?

Reply:
It's a bit convoluted, but I think it's handled gracefully: if `fScanningWallet` is set to `false` as `fAbortRescan` is being set to `true` (i.e. at the end of a scan), there is no issue as `ScanForWalletTransactions` sets `fAbortRescan` to `false` at its start.

BTW this behavior is unchanged - here's the prior implementation: https://github.com/bitcoin/bitcoin/pull/14942/files#diff-522490d83dce5375d423b23886e4125eL217-R219